server/webapp/WEB-INF/rails.new/webpack/models/preferences/filters_model.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+;(function () { // eslint-disable-line no-extra-semi
+  "use strict";
+
+  const m      = require("mithril");
+  const Stream = require("mithril/stream");
+
+  function FiltersModel(apiUrl, errors) {
+
+    const filters = Stream();
+
+    // persists checkbox state, defaulting to true. without this Stream,
+    // without this the checkbox will be reset each time a filter is created,
+    // which is an unexpected user experience.
+    const myCommits = Stream(true);
+
+    function fetchFilters() {
+      m.request({
```

Comment:
we use `$.ajax` instead of `m.request`.
Example: [`CRUD_mixin`](https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/webpack/models/mixins/crud_mixins.js)